src/libc_wrappers.c

```diff
@@ -13,31 +13,141 @@ Public License along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 */
 
+#define _GNU_SOURCE
```

Comment:
This is as much of a note to self as a review note: we should only use this in our internal headers, or undef this at end (doesn't matter for internal use though)